Python/ceval.c

```diff
@@ -1292,11 +1292,33 @@ eval_frame_handle_pending(PyThreadState *tstate)
     }
 
 #define CHECK_EVAL_BREAKER() \
+    CHECK_EMSCRIPTEN_SIGNALS(); \
```

Comment:
Why can't emscripten signals set the `eval_breaker` variable, like all other platforms?

Reply:
Because of the "shared-nothing" threading model. There is no preemptive multitasking so the thread that Python is running in can't be responsible for writing to the memory when Python code is running. The thread handling the UI cannot write into the Python/wasm memory because it doesn't have access to it. The embedder has to manually create a single byte of shared memory between the UI and the Python thread to handle the signal. But this byte lives outside the wasm memory and cannot be accessed via a pointer (so it can't be accessed from C at all).

Reply:
This seems odd.

So, you can read a shared buffer from javascript, but not web assembly? But you can call javascript from web assembly, though? So this is a pointless limitation of web assembly, as just impairs performance, but offers no additional security?

Reply:
> So this is a pointless limitation of web assembly, as just impairs performance, but offers no additional security?

Yes. There are plans to eventually lift this limitation some time in the next few years.
https://github.com/WebAssembly/multi-memory/blob/main/proposals/multi-memory/Overview.md
But we also need to get llvm + emscripten support, so even once wasm adds the feature it will take a while before it is accessible via our compiler toolchain.

Reply:
Sure. I just wanted to make sure I understood this correctly.

Reply:
To describe the situation more completely:
1. Memory buffers can either be marked shared or private
2. shared memory has worse performance than unshared memory because of mitigations for cache sidechannel attacks like Spectre and Heartbleed
3. wasm currently only allows one memory

For performance reasons, we want the one memory to be private. So we're stuck with what this PR does.

With multiple memories, we could have a private memory and a shared memory. We would need some attribute that asks the linker to allocate certain things in the shared memory, and then we could mark `eval_breaker` and similar signal flags with this "shared" attribute. In any case this is blocked on the wasm multiple memories proposal.